src/NBitcoin/Consensus.cs

```diff
@@ -116,10 +125,12 @@ public class Consensus : IConsensus
             int lastPowBlock,
             BigInteger proofOfStakeLimit,
             BigInteger proofOfStakeLimitV2,
-            Money proofOfStakeReward
-            )
+            Money proofOfStakeReward)
         {
-            this.Rules = new List<IConsensusRule>();
+            this.IntegrityValidationRules = new List<ISyncBaseConsensusRule>();
+            this.HeaderValidationRules = new List<ISyncBaseConsensusRule>();
+            this.PartialValidationRules = new List<IAsyncBaseConsensusRule>();
+            this.FullValidationRules = new List<IAsyncBaseConsensusRule>();
```

Comment:
If the member is IAsync... then maybe suffix the member with Async.  Like this.FullValidationRulesAsync and this.PartialValidationRulesAsync.

Reply:
@Aprogiena @dangershony I don't mind both ways. Should I rename?

Reply:
I don't care